packages/jsonc/jsonc.test.mjs

```diff
@@ -35,17 +43,49 @@ describe('jsonc loader', { concurrency: true }, () => {
 		});
 
 		it('should handle specifiers with appending data', async () => {
-			await assertSuffixedSpecifiers(resolve, './fixture.jsonc', 'jsonc', { importAttributes: { type: 'jsonc' } });
+			await assertSuffixedSpecifiers(resolve, './fixture.jsonc', 'jsonc', {
+				importAttributes: { type: 'jsonc' },
+			});
+		});
+
+		it('should ignore json files that aren’t jsonc', async () => {
+			const result = await resolve(
+				'./fixture.txt',
+				{ importAttributes: { type: 'jsonc' } },
+				nextResolve,
+			);
+
+			assert.deepEqual(result, {
+				format: 'unknown',
+				url: './fixture.txt',
+			});
+		});
```

Comment:
I think import attributes is supposed to be authoritative—when file extension and import attrs disagree, import attrs is supposed to win:

> Import attributes tell the runtime how a particular module should be loaded.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with

So in this case, `result.format` should be `jsonc`.

Reply:
so we just need to remove one condition:
```js
	if (ext === '.jsonc' && ctx.importAttributes.type === 'jsonc')
```